sql/src/main/java/io/crate/planner/node/management/ExplainPlan.java

```diff
@@ -177,35 +176,48 @@ private Row buildResponse(Map<String, Long> apeTimings, Map<String, Map<String,
                                                                                    Collection<NodeOperation> nodeOperations) {
         Set<String> nodeIds = NodeOperationGrouper.groupByServer(nodeOperations).keySet();
 
-        if (nodeIds.size() > 0) {
-            CompletableFuture<Map<String, Map<String, Long>>> resultFuture = new CompletableFuture<>();
-            TransportCollectProfileOperation collectProfileOperation = getTransportCollectProfileOperation(executor, jobId);
-
-            ConcurrentHashMap<String, Map<String, Long>> mergedMap = new ConcurrentHashMap<>(nodeIds.size());
-            AtomicInteger counter = new AtomicInteger(nodeIds.size());
-
-            for (String nodeId : nodeIds) {
-                collectProfileOperation.collect(nodeId)
-                    .whenComplete((map, throwable) -> {
-                        if (throwable == null) {
-                            mergedMap.put(nodeId, map);
-                            if (counter.decrementAndGet() == 0) {
-                                resultFuture.complete(mergedMap);
-                            }
-                        } else {
-                            resultFuture.completeExceptionally(throwable);
-                        }
-                    });
-            }
-            return resultFuture;
-        } else {
-            // Collect from local JobExecutionContext
-            return executor
+        CompletableFuture<Map<String, Map<String, Long>>> resultFuture = new CompletableFuture<>();
+        TransportCollectProfileOperation collectProfileOperation = getTransportCollectProfileOperation(executor, jobId);
+
+        ConcurrentHashMap<String, Map<String, Long>> mergedMap = new ConcurrentHashMap<>(nodeIds.size());
+        boolean needsCollectLocal = !nodeIds.contains(executor.localNodeId());
```

Comment:
The local node is never listed in the nodeIds? Would be nice to somehow have the same execution paths regardless of local or remote nodes.

Reply:
If the local node is a master or client only node, the node is node in the nodeIds.

Yes, a same execution path would be nice, but it's not possible (to what I understood) because the collection of the remote nodes requires the transport action and the local node collection is done by the node action.

Reply:
Couldn't the collection of the local information also be achieved using the same transport action? Would make this much simpler. If not possible, it's fine like it is.

Reply:
The `collectProfileOperation` is actually a `remoteCollectOperation` that creates a request and sends it via the transport protocol. I've renamed the variable as @mfussenegger suggested https://github.com/crate/crate/pull/7370#discussion_r192658515 to make it clearer.

Reply:
Is it possible to send this request also to the local node? That would eliminate the extra path for collecting the local timings.